subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/conflicts/FailOnVersionConflictArtifactsVisitor.java

```diff
@@ -96,7 +96,7 @@ public void finishArtifacts() {
     @Override
     public void complete() {
         if (!allConflicts.isEmpty()) {
-            throw new VersionConflictException(projectPath, configurationName, allConflicts);
+            throw VersionConflictException.create(projectPath, configurationName, allConflicts);
```

Comment:
💭 While you're working in this area, it might be worth changing to use a more descriptive type for `allConflicts` that just `Pair`, where you could reference the versions and message with getters named for them, instead of just with `left()` and `right()`.